lib/executor/instantiate/component/component_instance.cpp

```diff
@@ -68,78 +72,154 @@ Executor::instantiate(Runtime::StoreManager &StoreMgr,
                           Arg.getName());
             return Unexpect(ErrCode::Value::ComponentNotImplInstantiate);
-          case AST::Component::Sort::CoreSortType::Instance:
-            StoreMgr.addNamedModule(
-                Arg.getName(), CompInst.getModuleInstance(SortIdx.getIdx()));
-            break;
           default:
             assumingUnreachable();
           }
         } else {
           switch (Sort.getSortType()) {
           case AST::Component::Sort::SortType::Func:
+            ImportMgr.exportFunction(Arg.getName(),
+                                     CompInst.getFunction(SortIdx.getIdx()));
+            break;
+          case AST::Component::Sort::SortType::Instance:
+            ImportMgr.exportComponentInstance(
+                Arg.getName(), CompInst.getComponentInstance(SortIdx.getIdx()));
+            break;
           case AST::Component::Sort::SortType::Value:
           case AST::Component::Sort::SortType::Type:
-          case AST::Component::Sort::SortType::Instance:
+          case AST::Component::Sort::SortType::Component:
+            // TODO: COMPONENT - complete the instance instantiation.
             spdlog::error(ErrCode::Value::ComponentNotImplInstantiate);
             spdlog::error("    incomplete instantiate (with {})"sv,
                           Arg.getName());
             return Unexpect(ErrCode::Value::ComponentNotImplInstantiate);
-          case AST::Component::Sort::SortType::Component:
-            EXPECTED_TRY(
-                StoreMgr
-                    .registerComponent(
-                        Arg.getName(),
-                        CompInst.getComponentInstance(SortIdx.getIdx()))
-                    .map_error([](auto E) {
-                      spdlog::error(E);
-                      spdlog::error(
-                          "    failed to register component instance"sv);
-                      return E;
-                    }));
-            break;
           default:
             assumingUnreachable();
           }
         }
       }
-      EXPECTED_TRY(auto Inst,
-                   instantiate(StoreMgr, CompInst.getComponent(
-                                             Expr.getComponentIndex())));
-      CompInst.addComponentInstance(std::move(Inst));
+      const AST::Component::Component &Comp =
+          CompInst.getComponent(Expr.getComponentIndex());
+      EXPECTED_TRY(auto NewCompInst, instantiate(ImportMgr, Comp));
+      CompInst.addComponentInstance(std::move(NewCompInst));
     } else {
-      // TODO: complete inline exports
-      spdlog::error(ErrCode::Value::ComponentNotImplInstantiate);
-      return Unexpect(ErrCode::Value::ComponentNotImplInstantiate);
+      // Inline exports case.
+      // Create a component instance with the exports.
+      auto Comp = std::make_unique<Runtime::Instance::ComponentInstance>("");
+      uint32_t CoreExpIdx[7] = {0, 0, 0, 0, 0, 0, 0};
+      uint32_t ExpIdx[5] = {0, 0, 0, 0, 0};
```

Comment:
Maybe record meaning of number of each entry?

Reply:
Yes. This is a temporary record table for export indices.
They'll be moved into component instance / import manager in the final refining.